Scripts/compile_demo_apps.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# source: https://stackoverflow.com/a/3352015/4080860
+trim() {
+    local var="$*"
+    # remove leading whitespace characters
+    var="${var#"${var%%[![:space:]]*}"}"
+    # remove trailing whitespace characters
+    var="${var%"${var##*[![:space:]]}"}"   
+    echo -n "$var"
+}
+
+build() {
+    xcodebuild build                       \
+        -workspace Demos/Demos.xcworkspace \
+        -scheme "$1"                       \
+        -sdk iphonesimulator               \
+    | bundle exec xcpretty -c
+}
+
+# tail -n +3 removes the first 3 lines
+SCHEMES=$(xcodebuild -workspace Demos/Demos.xcworkspace/ -list | tail -n +3)
```

Comment:
Is it correct? Search for `Schemes:` as I suggest will be more safe. also, as I suppose, `head` can skip first lines.

Reply:
Also, how do you map scheme on its sdk/target? Not all schemes are for iphonesimulator. There are some schemes for macOS.

Reply:
xcodebuild handles it I believe. The commands we used earlier to build the demo projects didn't have a `-destination` specifier.

Reply:
Regarding `tail`, I think it's alright enough. It is correct the way it's implemented now. It's either that or parsing the json returned by `xcodebuild -workspace Demos/Demos.xcworkspace/ -list -json`.

Reply:
@hhanesand json is better. I caught strange output several times:

```
$ xcodebuild -workspace Demos.xcworkspace/ -list
2019-01-16 10:25:20.573 xcodebuild[71745:2759467]  DTDeviceKit: deviceType from 667f7238dd76a8f087e96e9cc171729a217cc561 was NULL
2019-01-16 10:25:20.688 xcodebuild[71745:2759464]  DTDeviceKit: deviceType from 667f7238dd76a8f087e96e9cc171729a217cc561 was NULL
Information about workspace "Demos":
    Schemes:
        BenchmarkIPhone
```